applications/mne_fiff_anonymizer/fiffanonymizerapp.h

```diff
@@ -0,0 +1,202 @@
+// DEFINE NAMESPACE
+//=============================================================================================================
+namespace MNEFIFFANONYMIZER
+{
+
+//=============================================================================================================
+/**
+* Description of what this class is intended to do (in detail).
+*
+* @brief FiffAnonymizerApp class declaration.
+*/
+
+class FiffAnonymizerApp
+{
+public:
+    const double version;
+    const double maxValidFiffVerion;
+    const QString versionStr;
+    const QString name;
+    const QString description;
+public:
+    //    /**
+    //    * Constructs a FiffAnonymizerApp object.
+    //    */
+    FiffAnonymizerApp();
+
+    //copy constructor
+    //move constructor
+    int anonymizeFile();
+    void setFileIn(QFile* file);
+    void setFileOut(QFile* file);
+    void setVerboseMode(bool v);
+    bool getVerboseMode();
+    void setBruteMode(bool b);
+    void setQuietMode(bool q);
+    void setMeasurementDay(QString d);
+    void setMeasurementDayOffset(int d);
+    void setMeasurementDayOffset(bool b);
+    void setSubjectBirthday(QString d);
+    void setSubjectBirthdayOffset(int d);
+    void setSubjectBirthdayOffset(bool b);
+
+private:
+    void updateBlockTypeList(FIFFLIB::FiffTag::SPtr pTag);
+    bool checkValidFiffFormatVersion(FIFFLIB::FiffTag::SPtr pTag);
+    int censorTag(FIFFLIB::FiffTag::SPtr pInTag,FIFFLIB::FiffTag::SPtr pOutTag);
+    void addEntryToDir(FIFFLIB::FiffTag::SPtr pTag,qint64 filePos);
+    void addFinalEntryToDir();
+    void dir2tag(FIFFLIB::FiffTag::SPtr pTag);
+    void writeDirectory(FIFFLIB::FiffStream* stream,FIFFLIB::fiff_long_t pos=-1);
+    void updatePointer(FIFFLIB::FiffStream* stream, FIFFLIB::fiff_int_t tagKind,FIFFLIB::fiff_long_t newPos);
+    bool checkDeleteInputFile();
+    void printIfVerbose(const QString str,bool sameLine=false);
+
+    bool m_bVerboseMode;
+    bool m_bBruteMode;
+    bool m_bQuietMode;
+    bool m_bDeleteInputFileAfter;
+    bool m_bDeleteInputFileConfirmation;
+
+    QString m_sDfltString;
+    QDateTime m_dateDfltDate;
+    int  m_iMeasurementDayOffset;
+    bool m_bUseMeasurementDayOffset;
+    int  m_iSubjectBirthdayOffset;
+    bool m_bUseSubjectBirthdayOffset;
+
+    QDateTime m_dateMeasurmentDate;
+    QDateTime m_date_IdDate;
+    QByteArray m_BDfltMAC;
+
+    int m_iDfltSubjectId;
+    QString m_sDfltSubjectFirstName;
+    QString m_sDfltSubjectMidName;
+    QString m_sDfltSubjectLastName;
+    QDateTime m_dateSubjectBirthDay;
+    int m_iDfltSubjectWeight;
+    int m_iDfltSubjectHeight;
+    QString m_sDfltSubjectComment;
+    QString m_sDfltSubjectHisId;
+
+    int m_iDfltProjectId;
+    QString m_sDfltProjectName;
+    QString m_sDfltProjectAim;
+    QString m_sDfltProjectPersons;
+    QString m_sDfltProjectComment;
+
+    QString m_sFileNameIn;
+    QString m_sFfileNameOut;
+
+    QFile* m_fFileIn;
+    QFile* m_fFileOut;
+
+    QDebug *m_printInSameLineHelper;
```

Comment:
Try to avoid raw pointers as much as possible. If you still need to use them make sure you handle the garbage collection, i.e. delete them in the class' destructor. I would recommend using a guarded pointer `QPointer<QDebug>`, see [docu](https://doc.qt.io/qt-5/qpointer.html#details). Another benefit when using `QPointer` is that initializes to `Q_NULLPTR` by default and when delete was called. This way checking the pointer integrity via `if(pointer){...}` will never crash.

Why is a pointer needed for this member? Why don't you use a simple `qDebug() << ..` in `printIfVerbose()`?

Reply:
You spotted a problem. Solved.
My intention is to be able to use qDebug and print to the same line. Say you want to print:
"Reading file..."
Then perform the reading and depending on the result have the user see:
"Reading file... OK" or  "Reading file... ERROR"
I store the "previously used QDebug object" in m_printInSameLineHelper, maybe I should've called it m_qdebugPreviousUseHelper. or something of the sort.

Reply:
Great read on logging in Qt is also available [here](https://www.kdab.com/wp-content/uploads/stories/slides/Day2/KaiKoehne_Qt%20Logging%20Framework%2016_9_0.pdf)

Reply:
Thank you for the read. That is great.
I'm a bit confused. I thought c++ people discourage the use of macros, and promote the use of enum class and const or constexpr for simple variables. I'm surprised how Qt actively uses them. Sometimes in mne-cpp you can see things like "return (expr);" which remind me of the problems with macros.